src/fields.rs

```diff
@@ -899,180 +945,211 @@ The result of transforming `value as U`. Where `U` is wider than `T`, this
 zero-extends; where `U` is narrower, it truncates.
 **/
 fn resize<T, U>(value: T) -> U
-where T: BitStore, U: BitStore {
-	let zero = 0usize;
-	let mut slab = zero.to_ne_bytes();
-	let start = 0;
-
-	/* Copy the source value into the correct region of the intermediate slab.
-
-	The `BitStore::as_bytes` method returns the value as native-endian-order
-	bytes. These bytes are then written into the correct location of the slab
-	(low addresses on little-endian, high addresses on big-endian) to be
-	interpreted as `usize`.
-	*/
-	match mem::size_of::<T>() {
-		n @ 1 | n @ 2 | n @ 4 | n @ 8 => {
-			#[cfg(target_endian = "big")]
-			let start = mem::size_of::<usize>() - n;
-
-			slab[start ..][.. n].copy_from_slice(value.as_bytes());
-		},
-		_ => unreachable!("BitStore is not implemented on types of this size"),
-	}
-	let mid = usize::from_ne_bytes(slab);
-	//  Truncate to the correct size, then wrap in `U` through the trait method.
-	match mem::size_of::<U>() {
-		1 => U::from_bytes(&(mid as u8).to_ne_bytes()[..]),
-		2 => U::from_bytes(&(mid as u16).to_ne_bytes()[..]),
-		4 => U::from_bytes(&(mid as u32).to_ne_bytes()[..]),
-		#[cfg(target_pointer_width = "64")]
-		8 => U::from_bytes(&mid.to_ne_bytes()[..]),
-		_ => unreachable!("BitStore is not implemented on types of this size"),
-	}
+where
+    T: BitStore,
+    U: BitStore,
+{
+    use core::intrinsics::copy_nonoverlapping;
+
+    let mut buf = U::all_bits_zeroes();
+    let size_of_from = mem::size_of::<T>();
+    let size_of_into = mem::size_of::<U>();
+    unsafe {
+        if cfg!(target_endian = "big") {
+            // Big endian fills from the end.
+            if size_of_from > size_of_into {
+                // T has more bytes, move its pointer to match the length of U.
+                copy_nonoverlapping(
+                    (&value as *const T as *const u8).add(size_of_from - size_of_into),
+                    &mut buf as *mut U as *mut u8,
+                    size_of_from,
```

Comment:
This should be `size_of_into`; otherwise, the copy overruns the `buf` value.

Reply:
Yeah, typed it wrong.